### PR TITLE
Using IIS modules UE pass

### PR DESCRIPTION
**Pull request by guardrex**
Addresses #5100 
Addresses #5135 

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/modules?branch=pr-en-us-5481)

A UE pass; however, @shirhatti would u like to take a peek to see if we need to update anything?

### CODE REVIEW

**Comment by guardrex** [comment]
@shirhatti Thanks.

Let's piggyback other inproc changes on [ANCM: RequestTimeout doesn't apply to inProcess (aspnet/Docs \#5449)](https://github.com/aspnet/Docs/issues/5449).

File: aspnetcore/host-and-deploy/iis/modules.md [hunk 1]
```diff
@@ -15,75 +15,75 @@ uid: host-and-deploy/iis/modules
 
 By [Luke Latham](https://github.com/guardrex)
 
-ASP.NET Core applications are hosted by IIS in a reverse proxy configuration. Some of the native IIS modules and all of the IIS managed modules aren't available to process requests for ASP.NET Core apps. In many cases, ASP.NET Core offers an alternative to the features of IIS native and managed modules.
+ASP.NET Core apps are hosted by IIS in a reverse proxy configuration. Some of the native IIS modules and all of the IIS managed modules aren't available to process requests for ASP.NET Core apps. In many cases, ASP.NET Core offers an alternative to the features of IIS native and managed modules.
 
 ## Native Modules
```
Comment (scottaddie) [inline aspnetcore/host-and-deploy/iis/modules.md]: Change title to "Native modules" while you're in here, please.

File: aspnetcore/host-and-deploy/iis/modules.md [hunk 2]
```diff
@@ -15,75 +15,75 @@ uid: host-and-deploy/iis/modules
...
+| **HTTP Redirection**<br>`HttpRedirectionModule` | Yes | [URL Rewriting Middleware](xref:fundamentals/url-rewriting) |
+| **IIS Client Certificate Mapping Authentication**<br>`IISCertificateMappingAuthenticationModule` | Yes | |
+| **IP and Domain Restrictions**<br>`IpRestrictionModule` | Yes | |
+| **ISAPI Filters**<br>`IsapiFilterModule` | Yes | [Middleware](xref:fundamentals/middleware/index) |
+| **ISAPI**<br>`IsapiModule` | Yes | [Middleware](xref:fundamentals/middleware/index) |
+| **Protocol Support**<br>`ProtocolSupportModule` | Yes | |
+| **Request Filtering**<br>`RequestFilteringModule` | Yes | [URL Rewriting Middleware `IRule`](xref:fundamentals/url-rewriting#irule-based-rule) |
+| **Request Monitor**<br>`RequestMonitorModule` | Yes | |
+| **URL Rewriting**<br>`RewriteModule` | Yes&#8224; | [URL Rewriting Middleware](xref:fundamentals/url-rewriting) |
+| **Server Side Includes**<br>`ServerSideIncludeModule` | No | |
```
Comment (scottaddie) [inline aspnetcore/host-and-deploy/iis/modules.md]: Add a hyphen between "Server" and "Side"

File: aspnetcore/host-and-deploy/iis/modules.md [hunk 3]
```diff
@@ -15,75 +15,75 @@ uid: host-and-deploy/iis/modules
...
+| **Static Compression**<br>`StaticCompressionModule` | No | [Response Compression Middleware](xref:performance/response-compression) |
+| **Static Content**<br>`StaticFileModule` | No | [Static File Middleware](xref:fundamentals/static-files) |
+| **Token Caching**<br>`TokenCacheModule` | Yes | |
+| **URI Caching**<br>`UriCacheModule` | Yes | |
+| **URL Authorization**<br>`UrlAuthorizationModule` | Yes | [ASP.NET Core Identity](xref:security/authentication/identity) |
+| **Windows Authentication**<br>`WindowsAuthenticationModule` | Yes | |
+
+&#8224;The URL Rewrite Module's `isFile` and `isDirectory` don't work with ASP.NET Core apps due to the changes in [directory structure](xref:host-and-deploy/directory-structure).
```
Comment (scottaddie) [inline aspnetcore/host-and-deploy/iis/modules.md]: I assume `isFile` and `isDirectory` are properties or attributes. I recommend using one of those words before "don't work" to add clarity.

File: aspnetcore/host-and-deploy/iis/modules.md [hunk 4]
```diff
@@ -15,75 +15,75 @@ uid: host-and-deploy/iis/modules
...
 
 ## Managed Modules
```
Comment (scottaddie) [inline aspnetcore/host-and-deploy/iis/modules.md]: Managed Modules --> Managed modules

File: aspnetcore/host-and-deploy/iis/modules.md [hunk 5]
```diff
@@ -15,75 +15,75 @@ uid: host-and-deploy/iis/modules
...
+| FileAuthorization       | No               | |
+| FormsAuthentication     | No               | [Cookie Authentication Middleware](xref:security/authentication/cookie) |
+| OutputCache             | No               | [Response Caching Middleware](xref:performance/caching/middleware) |
+| Profile                 | No               | |
+| RoleManager             | No               | |
+| ScriptModule-4.0        | No               | |
+| Session                 | No               | [Session Middleware](xref:fundamentals/app-state) |
+| UrlAuthorization        | No               | |
+| UrlMappingsModule       | No               | [URL Rewriting Middleware](xref:fundamentals/url-rewriting) |
+| UrlRoutingModule-4.0    | No               | [ASP.NET Core  Identity](xref:security/authentication/identity) |
```
Comment (scottaddie) [inline aspnetcore/host-and-deploy/iis/modules.md]: Remove the extra space in the "ASP.NET Core  Identity" link text.

File: aspnetcore/host-and-deploy/iis/modules.md [hunk 6]
```diff
@@ -15,75 +15,75 @@ uid: host-and-deploy/iis/modules
...
+| WindowsAuthentication   | No               | |
 
 ## IIS Manager application changes
 
-Using IIS Manager to configure settings, the *web.config* file of the app is changed. If deploying an app and including *web.config*, any changes made with IIS Manger are overwritten by the deployed *web.config* file. If changes are made to the server's *web.config* file, copy the updated *web.config* file to the local project immediately.
+When using IIS Manager to configure settings, the *web.config* file of the app is changed. If deploying an app and including *web.config*, any changes made with IIS Manger are overwritten by the deployed *web.config* file. If changes are made to the server's *web.config* file, copy the updated *web.config* file on the server to the local project immediately.
```
Comment (scottaddie) [inline aspnetcore/host-and-deploy/iis/modules.md]: Manger --> Manager

File: aspnetcore/host-and-deploy/iis/modules.md [hunk 7]
```diff
@@ -93,17 +93,19 @@ Many modules offer a configuration setting that allows them to be disabled them
...
+1. Unlock the **\<modules>** section of *web.config*. In the **Connections** sidebar, select the website in **Sites**. In the **Management** area, open the **Configuration Editor**. Use the navigation controls to select the `system.webServer/modules` section. In the **Actions** sidebar on the right, select to **Unlock** the section.
 
-4. At this point, a `<modules>` section can be added to the *web.config* file with a `<remove>` element to remove the module from the app. Multiple `<remove>` elements can be added to remove multiple modules. Don't forget that if *web.config* changes are made on the server to make them immediately in the project locally. Removing a module this way won't affect the use of the module with other apps on the server.
+1. At this point, a **\<modules>** section can be added to the *web.config* file with a **\<remove>** element to remove the module from the app. Multiple **\<remove>** elements can be added to remove multiple modules. Don't forget that if *web.config* changes are made on the server to make them immediately in the project locally. Removing a module this way won't affect the use of the module with other apps on the server.
```
Comment (scottaddie) [inline aspnetcore/host-and-deploy/iis/modules.md]: ~~Don't forget that if *web.config* changes are made on the server to make them immediately in the project locally.~~
If *web.config* changes are made on the server, make them immediately in the project locally.

File: aspnetcore/host-and-deploy/iis/modules.md (hunk 2 above)
Comment (guardrex) [reply aspnetcore/host-and-deploy/iis/modules.md]: The funny/sad thing is that MS is inconsistent about it. I think the dashed version edges out the version without a dash looking at MS blogs and MSDN (we're not helping tho with https://docs.microsoft.com/iis/configuration/system.webserver/serversideinclude, where the title is *Server Side Include \<serverSideInclude>*.

The OS isn't all that helpful. Although WinOS has it with a dash ...

capture1

WS 2016 doesn't ...

capture2

😭 ...... Let's go with the dash.

File: aspnetcore/host-and-deploy/iis/modules.md (hunk 3 above)
Comment (guardrex) [reply aspnetcore/host-and-deploy/iis/modules.md]: Good idea. They call them "match types" ... e.g. ...

```xml
<add input="{REQUEST_FILENAME}" matchType="isFile" negate="true">
```

File: aspnetcore/host-and-deploy/iis/modules.md (hunk 6 above)
Comment (guardrex) [reply aspnetcore/host-and-deploy/iis/modules.md]: :alien: ........... I did *not* do that. :smile:

File: aspnetcore/host-and-deploy/iis/modules.md (hunk 7 above)
Comment (guardrex) [reply aspnetcore/host-and-deploy/iis/modules.md]: Yep ... so much for conversational tone! :smile:

Comment (guardrex) [reply aspnetcore/host-and-deploy/iis/modules.md]: I'm going to take a stab at avoiding the pronoun with the slightly longer ...

> If *web.config* changes are made on the server, immediately make the same changes to the project's *web.config* file locally.

**Pull request merged**